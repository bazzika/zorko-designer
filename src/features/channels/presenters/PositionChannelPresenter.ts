import {PositionChannelState} from './PositionChannelState';
import {BaseStatePresenter} from '../../../core/BaseStatePresenter';

export class PositionChannelPresenter extends BaseStatePresenter<PositionChannelState> {
  static create(state?: PositionChannelState) {
    return new PositionChannelPresenter(state);
  }

  getDefaultState() {
    return {
      name: '',
      field: null,
      type: null
    };
  }

  setField(field) {
    this.state.field = field;
    return this;
  }

  setType(type: string): this {
    this.state.type = type;
    return this;
  }

  toState() {
    return this.state;
  }

  setName(name: string): this {
    this.state.name = name;
    return this;
  }

  getName() {
    return this.state.name;
  }

  getType() {
    return this.state.type;
  }

  getField() {
    return this.state.field;
  }
}
