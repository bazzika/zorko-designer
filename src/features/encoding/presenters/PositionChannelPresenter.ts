export type PositionChannelState = {
  name: string;
  field?: any;
  type?: any;
};

export class PositionChannelPresenter {
  private readonly state: PositionChannelState;

  static create(state?: PositionChannelState) {
    return new PositionChannelPresenter(state);
  }

  constructor(state?: PositionChannelState) {
    if (!state) {
      state = {name: 'x', field: null, type: null};
    }

    this.state = state;
  }

  toState() {
    return this.state;
  }
}
