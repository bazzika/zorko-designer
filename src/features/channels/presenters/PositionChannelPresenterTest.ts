import {PositionChannelPresenter} from './PositionChannelPresenter';

describe('PositionChannelPresenter', () => {
  let presenter: PositionChannelPresenter;

  beforeEach(() => {
    presenter = PositionChannelPresenter.create();
  });

  it('creates default state', () => {
    expect(PositionChannelPresenter.create().toState()).toMatchSnapshot();
  });

  it('sets field', () => {
    expect(presenter.setField('a').toState()).toMatchSnapshot();
  });

  it('sets type', () => {
    expect(presenter.setType('quantitative').toState()).toMatchSnapshot();
  });

  it('sets name', () => {
    expect(presenter.setName('x').toState()).toMatchSnapshot();
  });

  it('gets name', () => {
    expect(presenter.setName('x').getName()).toEqual('x');
  });

  it('gets type', () => {
    expect(presenter.setType('ordinal').getType()).toEqual('ordinal');
  });

  it('gets field', () => {
    expect(presenter.setField('a').getField()).toEqual('a');
  });
});
