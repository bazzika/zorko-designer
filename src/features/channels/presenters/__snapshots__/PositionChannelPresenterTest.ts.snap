// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionChannelPresenter creates default state 1`] = `
Object {
  "field": null,
  "name": "",
  "type": null,
}
`;

exports[`PositionChannelPresenter sets field 1`] = `
Object {
  "field": "a",
  "name": "",
  "type": null,
}
`;

exports[`PositionChannelPresenter sets name 1`] = `
Object {
  "field": null,
  "name": "x",
  "type": null,
}
`;

exports[`PositionChannelPresenter sets type 1`] = `
Object {
  "field": null,
  "name": "",
  "type": "quantitative",
}
`;
